scene/gui/tab_bar.cpp

```diff
@@ -824,22 +1030,68 @@ void TabBar::_update_cache() {
 		return;
 	}
 
-	Ref<StyleBox> tab_disabled = get_theme_stylebox(SNAME("tab_disabled"));
-	Ref<StyleBox> tab_unselected = get_theme_stylebox(SNAME("tab_unselected"));
-	Ref<StyleBox> tab_selected = get_theme_stylebox(SNAME("tab_selected"));
-	Ref<Texture2D> incr = get_theme_icon(SNAME("increment"));
-	Ref<Texture2D> decr = get_theme_icon(SNAME("decrement"));
+	Ref<StyleBox> tab_unselected = get_theme_stylebox(vertical_mode ? SNAME("vertical_tab_unselected") : SNAME("tab_unselected"));
+	Ref<StyleBox> tab_selected = get_theme_stylebox(vertical_mode ? SNAME("vertical_tab_selected") : SNAME("tab_selected"));
+	Ref<StyleBox> tab_disabled = get_theme_stylebox(vertical_mode ? SNAME("vertical_tab_disabled") : SNAME("tab_disabled"));
+	Ref<Texture2D> incr = get_theme_icon(vertical_mode ? SNAME("vertical_increment") : SNAME("increment"));
+	Ref<Texture2D> decr = get_theme_icon(vertical_mode ? SNAME("vertical_decrement") : SNAME("decrement"));
 
-	int limit = get_size().width;
-	int limit_minus_buttons = limit - incr->get_width() - decr->get_width();
+	int limit = vertical_mode ? get_size().height : get_size().width;
+	int limit_minus_buttons = vertical_mode ? limit - incr->get_height() - decr->get_height() : limit - incr->get_width() - decr->get_width();
 
-	int w = 0;
+	int ofs = 0;
 
 	max_drawn_tab = tabs.size() - 1;
 
 	for (int i = 0; i < tabs.size(); i++) {
 		tabs.write[i].text_buf->set_width(-1);
 		tabs.write[i].size_text = Math::ceil(tabs[i].text_buf->get_size().x);
+
+		if (vertical_mode) {
+			tabs.write[i].size_cache = get_tab_height(i);
+			int tab_width = get_tab_width(i);
+
+			if (max_width > 0 && tab_width > max_width) {
```

Comment:
I don't really see the point of setting a max width for vertical tabs. For horizontal tabs, it allows saving space, but this is not the case with vertical tabs.

Reply:
It should be most effective in containers since, otherwise the vertical TabBar will have its minimum size set to the longest tab, it doesn't support a specific overflow behavior, but maximum width allows you to control this otherwise. (and I believe you could feasibly implement overflow behavior with this implementation)

Reply:
I haven't tried it in a container, but at least with anchor/offset positioning, it didn't limit the actual tab width, but only the text width (no matter whether clipping was on or off). But if it works like you say, it's indeed useful to have.

Reply:
To be fair it might be more suitable to name another property and call it `tab_max_content_width` as it doesn't limit the right aligned elements, (since they don't follow the same content rules as the regular as since it was too ugly to be visually useful imo) but given `max_tab_width` is useless in vertical mode anyway, I had it perform that task, either renaming the property in the editor in vertical mode or using another property would likely be just as suitable.